File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/emptysubmission/EmptySubmissionHandler.java
package uk.nhs.nhsx.emptysubmission;

import uk.nhs.nhsx.core.Environment;
import uk.nhs.nhsx.core.HttpResponses;
import uk.nhs.nhsx.core.auth.ApiName;
import uk.nhs.nhsx.core.auth.Authenticator;
import uk.nhs.nhsx.core.routing.Routing;
import uk.nhs.nhsx.core.routing.RoutingHandler;

import static uk.nhs.nhsx.core.auth.StandardAuthentication.awsAuthentication;
import static uk.nhs.nhsx.core.routing.Routing.Method.POST;
import static uk.nhs.nhsx.core.routing.Routing.path;
import static uk.nhs.nhsx.core.routing.Routing.routes;
import static uk.nhs.nhsx.core.routing.StandardHandlers.withoutSignedResponses;

public class EmptySubmissionHandler extends RoutingHandler {

    private final Routing.Handler handler;

    @SuppressWarnings("unused")
    public EmptySubmissionHandler() {
        this(Environment.fromSystem());
    }

    public EmptySubmissionHandler(Environment environment) {
        this(environment, awsAuthentication(ApiName.Mobile));
    }

    public EmptySubmissionHandler(Environment environment,
                                  Authenticator authenticator) {
        handler = withoutSignedResponses(
            environment,
            authenticator,
            routes(
                path(POST, "/submission/empty-submission", (r) ->
                    HttpResponses.ok()
                )
            )
        );
    }

    @Override
    public Routing.Handler handler() {
        return handler;
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/emptysubmission/EmptySubmissionHandler.kt
package uk.nhs.nhsx.emptysubmission

import uk.nhs.nhsx.core.Environment
import uk.nhs.nhsx.core.HttpResponses
import uk.nhs.nhsx.core.SystemClock
import uk.nhs.nhsx.core.auth.ApiName.Mobile
import uk.nhs.nhsx.core.auth.Authenticator
import uk.nhs.nhsx.core.auth.StandardAuthentication.awsAuthentication
import uk.nhs.nhsx.core.events.Events
import uk.nhs.nhsx.core.events.PrintingJsonEvents
import uk.nhs.nhsx.core.handler.ApiGatewayHandler
import uk.nhs.nhsx.core.routing.Routing.Method.POST
import uk.nhs.nhsx.core.routing.Routing.path
import uk.nhs.nhsx.core.routing.Routing.routes
import uk.nhs.nhsx.core.handler.RoutingHandler
import uk.nhs.nhsx.core.routing.authorisedBy
import uk.nhs.nhsx.core.routing.withoutSignedResponses

@Suppress("unused")
class EmptySubmissionHandler @JvmOverloads constructor(
    environment: Environment = Environment.fromSystem(),
    events: Events = PrintingJsonEvents(SystemClock.CLOCK),
    authenticator: Authenticator = awsAuthentication(Mobile, events)
) : RoutingHandler() {

    private val handler: ApiGatewayHandler = withoutSignedResponses(
        events,
        environment,
        routes(
            authorisedBy(
                authenticator,
                path(POST, "/submission/empty-submission",
                    ApiGatewayHandler { _, _ -> HttpResponses.ok() })
            )
        )
    )

    override fun handler() = handler
}
